OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -50,21 +62,18 @@ public FrozenActor(Actor self, PPos[] footprint, Shroud shroud)
 			removeFrozenActors = self.TraitsImplementing<IRemoveFrozenActor>().ToArray();
 
 			// Consider all cells inside the map area (ignoring the current map bounds)
-			Footprint = footprint
-				.Where(m => shroud.Contains(m))
-				.ToArray();
+			Footprint = footprint.Where(shroud.Contains).ToArray();
 
 			CenterPosition = self.CenterPosition;
 			Bounds = self.Bounds;
 
+			var mod = self.TraitsImplementing<IRadarColorModifier>().FirstOrDefault();
+			RadarColor = mod != null ? mod.RadarColorOverride(self) : self.Owner.Color.RGB;
+			OccupiedCells = new HashSet<CPos>(footprint.SelectMany(uv => self.World.Map.Unproject(uv).Select(m => m.ToCPos(self.World.Map))));
```

Comment:
Does this have to be a set? Might be more efficient to do ```.Distinct.ToArray()``` (unless we need to do set queries on this anywhere, but I don't see any).